### PR TITLE
Set captain´s name from local storage

### PR DESCRIPTION
**Pull request by BelaM80**
I need a hint, how I can get rid of the quotationmarks around the name on the second page.

### CODE REVIEW

File: client/src/components/pages/StartPage.js [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
-import React from "react";
+import React, { useEffect, useState } from "react";
 import Container from "../Container";
 
 function StartPage() {
+  const [value, setValue] = useState(localStorage.getItem("name") || "");
+  useEffect(() => {
+    localStorage.setItem("name", value);
+  }, [value]);
+  const onChange = (event) => setValue(event.target.value);
```
Comment (jamarob) [inline client/src/components/pages/StartPage.js:11]: Think about pulling the state up into `App.js`, from there you can pass it down to `HomePage` and  `StartPage` via props, this is a general pattern in react.

As for the extra quotation marks around name alsways use `JSON.stringify` when saving and `JSON.parse` when loading data from `localStorage`. 

Beware, `JSON.parse` will throw an error if it encounters something funny (it is a little bit picky about what to read). Data passed through `JSON.stringify` will alwys be parsed correctly, though. Best practive is to wrap calls to `JSON.parse` with a `try-catch`-statement.

File: client/src/components/pages/StartPage.js [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
   return (
     <>
       <Header />
       <Container>
-        <p>Captain xy is required on bridge!</p>
+        <p onChange={onChange}>Captain {value} is required on bridge!</p>
```
Comment (jamarob) [inline client/src/components/pages/StartPage.js:16]: `<p>` will not fire `onChange` as it is readonly 😉

File: client/src/components/pages/HomePage.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ function HomePage() {
   const history = useHistory();
   const handleSubmit = (e) => {
     e.preventDefault();
-    const newName = [name];
+    const newName = name;
     setName("");
     localStorage.setItem("name", JSON.stringify(newName));
     history.push("/startpage");
```
Comment (jamarob) [inline client/src/components/pages/HomePage.js:14]: I think your state belongs in `App.js`, see comment in `StartPage`

**Review by jamarob — COMMENTED** [review]
Spacey, I like the look of it 🚀 

Left some comments regarding your question. As a general remark communicate via shared state from a parent component and put your load & save logic where the state is.

File: client/src/components/pages/StartPage.js (hunk 1 above)
Comment (BelaM80) [reply client/src/components/pages/StartPage.js:11]: Thanks for review! 👍🏻 🙏🏻

**Pull request merged**